specs/interop.md

```diff
@@ -0,0 +1,708 @@
+stagnant, all representations of chain ids MUST be the `uint256` type.
+
+In the future, OP Stack chains reserve the right to use up to 32 bytes to represent a chain id. The
+configuration of the chain should deterministically map to a chain id and with careful architecture
+changes, all possible OP Stack chains in the superchain will be able to exist counterfactually.
+
+It is a known issue that not all software in the Ethereum space can handle 32 byte chain ids.
+
+## Interop Network Upgrade
+
+The interop network upgrade timestamp defines the timestamp at which all functionality in this document is considered
+the consensus rules for an OP Stack based network. On the interop network upgrade block, a set of deposit transaction
+based upgrade transactions are deterministically generated by the derivation pipeline in the following order:
+
+- L1 Attributes Transaction calling `setL1BlockValuesEcotone`
+- User deposits from L1
+- Network Upgrade Transactions
+  - L1Block deployment
+  - CrossL2Inbox deployment
+  - L2ToL2CrossDomainMessenger deployment
+  - Update L1Block Proxy ERC-1967 Implementation Slot
+  - Update CrossL2Inbox Proxy ERC-1967 Implementation Slot
+  - Update L2ToL2CrossDomainMessenger Proxy ERC-1967 Implementation Slot
+  - Mint Cross Chain Ether Liquidity in L2ToL2CrossDomainMessenger
+
+The execution payload MUST set `noTxPool` to `true` for this block.
+
+The exact definitions for these upgrade transactions are still to be defined.
+
+## Derivation Pipeline
+
+The derivation pipeline enforces invariants on safe blocks that include executing messages.
+
+- The executing message MUST have a corresponding initiating message
+- The initiating message that corresponds to an executing message MUST come from a chain in its dependency set
+
+Blocks that contain transactions that relay cross domain messages to the destination chain where the
+initiating transaction does not exist MUST be considered invalid and MUST not be allowed by the
+derivation pipeline to be considered safe.
+
+There is no concept of replay protection at the lowest level of abstractions within the protocol because
+there is no replay protection mechanism that fits well for all applications. Users MAY submit an
+arbitrary number of executing messages per initiating message. Applications MUST build their own replay
+protection mechanisms if they are interacting with the lowest level abstractions.
+
+### Depositing an Executing Message
+
+Deposit transactions (force inclusion transactions) give censorship resistance to layer two networks.
+The derivation pipeline must gracefully handle the case in which a user uses a deposit transaction to
+relay a cross chain message. To not couple preconfirmation security to consensus, deposit transactions
+that execute cross chain messages MUST have an initiating message that is considered [safe](#safety) by the remote
+chain's derivation pipeline. This means that the remote chain's data including the initiating message
```

Comment:
Should this say "source chain" instead of "remote chain"? I see remote chain used in a few other places too, if those are all intended we should add remote chain to the definitions table above

Reply:
Yes good catch, will need to do a pass and replace the old language everywhere